src/test/java/com/skalashynski/spring/springboot/service/StudentServiceTest.java

```diff
@@ -0,0 +1,177 @@
+package com.skalashynski.spring.springboot.service;
+
+import com.skalashynski.spring.springboot.entity.Student;
+import com.skalashynski.spring.springboot.exception.ApiException;
+import com.skalashynski.spring.springboot.repository.StudentRepository;
+import com.skalashynski.spring.springboot.service.impl.StudentServiceImpl;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentMatchers;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.*;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.BDDMockito.*;
+
+@ExtendWith(MockitoExtension.class)
+class StudentServiceTest {
+    private StudentService studentService;
+    @Mock
+    private StudentRepository studentRepository;
+
+    private Student student;
+
+    @BeforeEach
+    void init() {
+        studentService = new StudentServiceImpl(studentRepository);
```

Comment:
I'm not sure if you need this initialisation. Spring may create this class itself. Please, make sure and adjust or convict me.

Reply:
Maybe I'm wrong, but I think that to use @Autowired I need to have the Spring context. And since I am writing a unit test and not an integration test, this is not necessary. I did this way
`@InjectMocks
    private StudentServiceImpl studentService;`
And I don't know if I'm doing the right thing by writing a test for an interface and linking it to a specific implementation.

Reply:
you are right, agree.
I did't mention you are testing only service logic.